server/src/services/search.service.ts

```diff
@@ -113,14 +113,25 @@ export class SearchService extends BaseService {
     }
 
     const userIds = this.getUserIdsToSearch(auth);
-    const key = machineLearning.clip.modelName + dto.query + dto.language;
-    let embedding = this.embeddingCache.get(key);
-    if (!embedding) {
-      embedding = await this.machineLearningRepository.encodeText(machineLearning.urls, dto.query, {
-        modelName: machineLearning.clip.modelName,
-        language: dto.language,
-      });
-      this.embeddingCache.set(key, embedding);
+    let embedding;
+    if (dto.query) {
+      const key = machineLearning.clip.modelName + dto.query + dto.language;
+      embedding = this.embeddingCache.get(key);
+      if (!embedding) {
+        embedding = await this.machineLearningRepository.encodeText(machineLearning.urls, dto.query, {
+          modelName: machineLearning.clip.modelName,
+          language: dto.language,
+        });
+        this.embeddingCache.set(key, embedding);
+      }
+    } else if (dto.queryAssetId) {
```

Comment:
You need to validate this, probably with requireAccess(Permission.AssetRead) or whatever.

Reply:
Is that not already being handled by the `@Authenticated({ permission: Permission.AssetRead })` up in [the controller layer function](https://github.com/immich-app/immich/blob/main/server/src/controllers/search.controller.ts#L58) that calls this function?

Or should I say, to have gotten this far, have we not already validated the caller has `Permission.AssetRead`?

Reply:
They do, yes. That validates the API key or session token has that permission, but it does not validate or imply that queryAssetId is an asset they should have access to. There is a difference between being able to read your assets vs all/any assets in the system.

Reply:
Ooooh neat didn't even know that mechanism existed

Implemented 👍